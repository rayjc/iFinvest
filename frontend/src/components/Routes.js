import React from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';
import PrivateRoute from './PrivateRoute';
import Auth from './Auth';
import Profile from './Profile';


const Routes = () => (
  <Switch>
    <PrivateRoute path="/portfolios/:id">
      <p>Investment forms and chart under this portfolio</p>
    </PrivateRoute>
    <Route exact path="/auth" >
      <Auth />
    </Route>
    <PrivateRoute exact path="/profile">
      <Profile />
    </PrivateRoute>
    <Route exact path="/">
      {/* <Home /> */}
      <p>Home page</p>
    </Route>
    <Redirect to="/" />
  </Switch>
);


export default Routes;