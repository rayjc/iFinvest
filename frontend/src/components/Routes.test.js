import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import Routes from './Routes';
import { useSelector, useDispatch } from 'react-redux';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch
}));

beforeEach(function() {
  useSelector.mockImplementation((selectorFn) =>
    selectorFn({
      user: {
        user: {
          id: 0, username: "test", first_name: "test", last_name: "user",
          email: "test@test.com"
        },
        error: null, isFetching: false,
      },
      portfolios: {
        portfolios: [
          { id: 0, name: "alpha", user_id: 0, created_at: "2020-01-10", investments: [] },
        ],
        error: null, isFetching: false,
      },
      investments: {
        investments: {}, error: null, isFetching: false,
      }
    }
    )
  );
});


// smoke test
it("renders without crashing", function() {
  render(<MemoryRouter><Routes /></MemoryRouter>);
});

// snapshot test
it("matches snapshot", function() {
  const { asFragment } = render(<MemoryRouter><Routes /></MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
});