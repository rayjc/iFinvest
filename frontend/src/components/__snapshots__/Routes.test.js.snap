// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="PortfolioList"
>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
    style="text-align: center;"
  >
    <div
      class="MuiBox-root MuiBox-root-5"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-root-6 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiCardActionArea-root makeStyles-actionArea-7"
              href="/portfolios/0"
              role="button"
              tabindex="0"
            >
              <h3
                class="MuiTypography-root MuiTypography-h5"
              >
                alpha
              </h3>
              <p
                class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom"
              >
                Jan 10th, 2020
              </p>
              <div
                class="MuiCardContent-root"
              />
              <span
                class="MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <div
              class="MuiCardActions-root MuiCardActions-spacing"
              style="justify-content: flex-end;"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
